rotkehlchen/tests/unit/decoders/test_pendle.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.evm.decoding.constants import CPT_GAS
+from rotkehlchen.chain.evm.decoding.pendle.constants import CPT_PENDLE
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.evm_event import EvmEvent
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.tests.utils.ethereum import get_decoded_events_of_transaction
+from rotkehlchen.types import Location, TimestampMS, deserialize_evm_tx_hash
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('ethereum_accounts', [['0xFd83CCCecef02a334e6A86e7eA8D0aa0F61f1Faf']])
+def test_lock_pendle(ethereum_inquirer, ethereum_accounts):
+    tx_hash = deserialize_evm_tx_hash('0xc8b252de1a62daa57d4fe294f371e67550e087fdeffe972261e1acc890d84bd5')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(evm_inquirer=ethereum_inquirer, tx_hash=tx_hash)
+    user_address, timestamp, gas_amount, approval_amount, out_amount, in_amount, locked_amount = ethereum_accounts[0], TimestampMS(1742478515000), '0.00118595784570676', '11.106239093069566243', '0.003319967510372966', '0.000065097402164175', '1110.62390930695662426'  # noqa: E501
+    expected_events = [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            amount=FVal(gas_amount),
+            location_label=user_address,
+            notes=f'Burn {gas_amount} ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_ETH,
+            amount=FVal(in_amount),
+            location_label=user_address,
+            notes=f'Receive {in_amount} ETH from 0x4f30A9D41B80ecC5B94306AB4364951AE3170210',
+            address=string_to_evm_address('0x4f30A9D41B80ecC5B94306AB4364951AE3170210'),
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=2,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_ETH,
+            amount=FVal(out_amount),
+            location_label=user_address,
+            notes=f'Send {out_amount} ETH to 0x4f30A9D41B80ecC5B94306AB4364951AE3170210',
+            address=string_to_evm_address('0x4f30A9D41B80ecC5B94306AB4364951AE3170210'),
```

Comment:
what are those eth events? are they some kind of payment or reward?

Reply:
> what are those eth events? are they some kind of payment or reward?

no. it's related to layerzero crosschain messaging. the state is broadcasted to all supported chains.

Reply:
Then this needs to be decoded properly. seems taht the user spends the ETH and it should be properly decoded as part of the lock logic. I guess you send X eth and you receive (X - used) ETH